daemon/algod/api/server/v2/utils.go

```diff
@@ -266,6 +266,30 @@ func convertToDeltas(txn node.TxnWithStatus) (*[]generated.AccountStateDelta, *g
 	return localStateDelta, stateDeltaToStateDelta(txn.ApplyData.EvalDelta.GlobalDelta)
 }
 
+func convertToLogItems(txn node.TxnWithStatus) *[]generated.LogItem {
+	var logItems *[]generated.LogItem
+	if len(txn.ApplyData.EvalDelta.Logs) > 0 {
+		l := make([]generated.LogItem, 0)
+
+		for _, v := range txn.ApplyData.EvalDelta.Logs {
+			// Resolve appid from index
+			var appid uint64
+			if v.ID == 0 {
+				appid = uint64(txn.Txn.Txn.ApplicationID)
```

Comment:
This will work for most app calls, but it won't for an app creation call, since `txn.Txn.Txn.ApplicationID` will be 0.

To fix it, you could make this function take an additional `*uint64` param which you use to pass in `response.ApplicationIndex` from `daemon/algod/api/server/v2/handlers.go`. Then if `txn.Txn.Txn.ApplicationID` is 0 and the additional `*uint64` param is not nil, assign it to `appid` instead.